driver/generic/xpt2046.py

```diff
@@ -0,0 +1,103 @@
+import time
+import machine
+import struct
+
+# see e.g.
+# https://github.com/MatthewLowden/RPi-XPT2046-Touchscreen-Python/blob/master/XPT2046.py
+# and the XPT2046 datasheet
+
+class Xpt2046:
+    CHAN_X  = const(0b0101_0000)
+    CHAN_Y  = const(0b0001_0000)
+    CHAN_Z1 = const(0b0011_0000)
+    CHAN_Z2 = const(0b0100_0000)
+    CHAN_T0 = const(0b0000_0000)
+    CHAN_T1 = const(0b0111_0000)
+    CHAN_BAT= const(0b0010_0000)
+    CHAN_AUX= const(0b0110_0000)
+
+    CONV_8_BIT =const(0b0000_1000)
+    CONV_12_BIT=const(0b0000_0000)
+    START_BIT  =const(0b1000_0000)
+
+    def _chanRead(self,chan):
+        self.cs.value(0)
+        struct.pack_into('BBB',self.buf,0,Xpt2046.START_BIT|self.conv|chan,0,0)
+        self.spi.write_readinto(self.buf,self.buf)
+        if self.conv==Xpt2046.CONV_8_BIT: ret=self.buf[1]
+        else: ret=(self.buf[1]<<4)|(self.buf[2]>>4)
+        self.cs.value(1)
+        return ret
+
+    def __init__(self,*,
```

Comment:
I think that `__init__` should also register this as an LVGL indev driver, instead of expecting the user to do it.